project_structure.txt

```diff
@@ -0,0 +1,119 @@
+.modin/
+   .config/
+   .distributed/
+   .pandas/
+   .test/
+   .rest files
+   .core/
+      .io/
+         .column_stores/
+            . # the same as it is now
+         .sql/
+            . # the same as it is now
+         .text/
+            . # the same as it is now
+         .file_dispatcher.py # the same as it is now
+         .io.py # the same as it is now
+      .dispatching/
+         .factories/
+            . # the same as it is now
+      .formats/
+         .base/
+            .query_compiler.py # BaseQueryCompiler
+         .pandas/
+            .base/ # or default (DefaultPandasQueryCompiler)
+               .query_compiler.py # BasePandasQueryCompiler
+            .kernels.py
+            .query_compiler.py # PandasQueryCompiler
+            .parsers.py
+            .utils.py # data_management/utils moved here
+      .execution/
+        .ray/
+           .frame/
+              .dataframe.py # GenericRayDataframe
+              .metadata.py # GenericRayDataframeMetadata
+           .partition_management/ # or `partitioning`
+              .axis_partition.py # GenericRayDataframeAxisPartition
+              .partition_manager.py # GenericRayDataframeManager
+              .partition.py # GenericRayDataframePartition
+           .io/
+              .io.py # GenericRayIO
+           .task_wrapper.py # GenericRayTask
+           .utils.py # init_ray
+        .dask/
+           . # the same as for ray
+        .python/
+           . # the same as for ray
+      .dataframe/  # implementations go here
+         .algebra/
+            .operators/ # dispatching/functions here
+               . # the same as it is in dispatching/functions
+         .implementations/ # all of the implementations go here
+            .base/
+               .frame/
```

Comment:
I have feeling there is an ambiguity between frame and dataframe terms here? Shouldn't we align these somehow?

Reply:
Yes, good point. `dataframe` is a better way to put it.